src/compiler/statement/ClausedNode.ts

```diff
@@ -0,0 +1,78 @@
+import * as ts from "typescript";
+import {Constructor} from "./../../Constructor";
+import * as errors from "./../../errors";
+import {verifyAndGetIndex, removeClausedNodeChildren} from "./../../manipulation";
+import {TypeGuards} from "./../../utils";
+import {Node} from "./../common";
+import {VariableDeclarationType} from "./VariableDeclarationType";
+
+export type ClausedNodeExtensionType = Node<ts.CaseBlock>;
+
+export interface ClausedNode {
```

Comment:
Does this need to be a mixin? Maybe we should just move all this directly onto `CaseBlock` then make these methods also available on `SwitchStatement`? I can't think of anything else in the language that would need this.